### PR TITLE
feat(lambda): Add EFS Filesystem support

### LINKED ISSUE #8595: AWS Lambda with EFS Filesystem support
New – A Shared File System for Your Lambda Functions
https://aws.amazon.com/tw/blogs/aws/new-a-shared-file-system-for-your-lambda-functions/

PR draft underway.

### Use Case

Running Lambda function with EFS Filesystem.

### Other

* [x] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by pahud**
feat(lambda): Add EFS Filesystem support

This PR adds `filesystemConfigs` construct property for `lambda.Function`.

### Considerations:
- [x] support efs filesystem at this moment but could support different types of filesystem in the future with the new `LambdaFileSystem` union-like class
- [ ] ensure all the selected vpc subnets for the lambda function have available efs mount target within the same subnet
- [ ] ensure the `efs.FileSystem` as well as its mount targets are ready before creating the lambda.Function
- [x] create the new `AWS::EFS::AccessPoint` resource under the hood 

Close #8595 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by jeznag** [comment]
Eagerly awaiting this :D

File: packages/@aws-cdk/aws-efs/lib/efs-file-system.ts [hunk 1]
```diff
@@ -217,6 +217,16 @@ export class FileSystem extends Resource implements IFileSystem {
...
 
+  /**
+   * The subnets of the mount targets
+   */
+  public readonly mountTargetSubnets: ec2.SelectedSubnets;
+
+  /**
+   * The mount targets
+   */
+  public readonly mountTargets: CfnMountTarget[];
```
Comment (eladb) [inline packages/@aws-cdk/aws-efs/lib/efs-file-system.ts:228]: We do not expose L2s in L1 APIs

Comment (pahud) [reply packages/@aws-cdk/aws-efs/lib/efs-file-system.ts:228]: > We do not expose L2s in L1 APIs

No problem. Let me nail it! 😄

**Comment by pahud** [comment]
oops! Looks like I deleted this PR by mistake. I'll figure it out and open again.

**Comment by pahud** [comment]
new PR  https://github.com/aws/aws-cdk/pull/8602

**Pull request closed without merging**